pkg/devfile/parser/writer_test.go

```diff
@@ -17,67 +17,96 @@ package parser
 
 import (
 	"fmt"
+	"strings"
+	"testing"
+
 	v1 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
 	apiAttributes "github.com/devfile/api/v2/pkg/attributes"
 	devfilepkg "github.com/devfile/api/v2/pkg/devfile"
 	devfileCtx "github.com/devfile/library/v2/pkg/devfile/parser/context"
 	v2 "github.com/devfile/library/v2/pkg/devfile/parser/data/v2"
 	"github.com/devfile/library/v2/pkg/testingutil/filesystem"
-	"strings"
-	"testing"
 )
 
-func TestWriteYamlDevfile(t *testing.T) {
+func TestDevfileObj_WriteYamlDevfile(t *testing.T) {
 
 	var (
 		schemaVersion = "2.2.0"
-		testName      = "TestName"
 		uri           = "./relative/path/deploy.yaml"
 		uri2          = "./relative/path/deploy2.yaml"
-		attributes    = apiAttributes.Attributes{}.PutString(K8sLikeComponentOriginalURIKey, uri)
-		attributes2   = apiAttributes.Attributes{}.PutString(K8sLikeComponentOriginalURIKey, uri2)
 	)
 
-	t.Run("write yaml devfile", func(t *testing.T) {
-
-		// Use fakeFs
-		fs := filesystem.NewFakeFs()
+	tests := []struct {
+		name     string
+		fileName string
+		wantErr  bool
+	}{
+		{
+			name:     "write devfile with .yaml extension",
+			fileName: OutputDevfileYamlPath,
+		},
+		{
+			name:     "write .devfile with .yaml extension",
+			fileName: ".devfile.yaml",
+		},
+		{
+			name:     "write devfile with .yml extension",
+			fileName: "devfile.yml",
+		},
+		{
+			name:     "write .devfile with .yml extension",
+			fileName: ".devfile.yml",
+		},
+		{
+			name:     "write any file, regardless of name and extension",
```

Comment:
this is out-of-scope for the current issue

Reply:
Actually, part of the acceptance criteria is for the library to work generically and support any file name (not just `.yml`)..